src/UI/Implementation/Component/Button/Button.php

```diff
@@ -0,0 +1,79 @@
+<?php
+
+/* Copyright (c) 2016 Richard Klees <richard.klees@concepts-and-training.de> Extended GPL, see docs/LICENSE */
+
+namespace ILIAS\UI\Implementation\Component\Button;
+
+use ILIAS\UI\Component as C;
+use ILIAS\UI\Implementation\Component\ComponentHelper;
+use ILIAS\UI\Component\Glyph\Glyph;
+
+/**
+ * This implements commonalities between standard and primary buttons. 
+ */
+abstract class Button implements C\Button\Button {
+	use ComponentHelper;
+
+	/**
+	 * @var string
+	 */
+	protected $label;
+
+    /**
+     * @var string
+     */
+    protected $action;
+
+    /**
+     * @var bool
+     */
+    protected $active = true;
+
+
+	public function __construct($label, $action) {
+        $this->checkStringArg("label", $label);
+        $this->checkStringArg("action", $action);
+		$this->label = $label;
+        $this->action = $action;
+	} 
+
+	/**
+	 * @inheritdocs 
+	 */
+	public function getLabel() {
+        return $this->label;
+	}
+
+	/**
+	 * @inheritdocs 
+	 */
+	public function withLabel($label) {
+        $this->checkStringArg("label", $label);
+        $clone = clone $this;
+        $clone->label = $label;
+        return $clone;
+	}
+
+	/**
+	 * @inheritdocs 
+	 */
+	public function getAction() {
+        return $this->action;
+	}
+
+	/**
+	 * @inheritdocs 
+	 */
+	public function isActive() {
+        return $this->active;
+	}
+
+	/**
+	 * @inheritdocs 
+	 */
+	public function withUnavailableAction() {
+        $clone = clone $this;
+        $clone->active = false;
+        return $clone;
+	}
+}
```

Comment:
This sounds good. However, we currently do not have this abstraction. 

Which way would you go: 

1. Not using JS on Buttons as long as the abstraction is not here (continue using old instances up to this point) 

2. OR introduce ID on buttons and begin using JS on Buttons through the approach described above now and introduce the abstraction later?